### ISSUE TITLE
showOpenDialog() filters showing extensions instead of names

### ISSUE THREAD

**Issue by chrilith** [electron/electron#12571]
* Electron version: 2.0.0-beta.X (tested on beta.1 and 7, not available in 1.8.4)
* Operating system: macOS High Sierra

### Expected behavior

When using `filters` option, `showOpenDialog()` should display a list of defined names.
<img width="533" alt="screen shot 2018-04-09 at 18 46 18" src="https://user-images.githubusercontent.com/1277891/38510823-79a4a5ae-3c26-11e8-9653-1357dedaf3f6.png">

### Actual behavior

When using `filters` option, `showOpenDialog()` shows only a list of extensions instead of the names.
<img width="569" alt="screen shot 2018-04-09 at 18 48 11" src="https://user-images.githubusercontent.com/1277891/38510873-9a49174a-3c26-11e8-80d6-2377c346c30f.png">

### How to reproduce

Tested only on macOS.

```javascript
const files = dialog.showOpenDialog({
	properties: ["openFile"],
	filters: [
		{ name: "Interchange File Format", extensions: ["lbm", "iff", "bl1", "bl2", "bl3"] },
		{ name: "Atari Degas Elite", extensions: ["pi1", "pi2", "pi3", "pc1", "pc2", "pc3"] }
	]
});
```

Let me know if you want a running project with this call!
Thanks!

**Comment by codebytere** [comment]
I believe actually that [this](https://github.com/electron/electron/pull/11959) PR fixes that, and it will be out with the next release.

**Comment by codebytere** [comment]
I believe actually that [this](https://github.com/electron/electron/pull/11959) PR fixes that, and it will be out with the next release.

**Comment by chrilith** [comment]
Indeed! Let's wait for the next version to see if #11959 will fix this.

Thanks!

**Comment by codebytere** [comment]
I'm gonna close this as resolved in the meantime, but let me know if it doesn't work as intended and i'll reopen it :)

**Comment by chrilith** [comment]
Tried 2.0.0-beta.8, still the same problem.

**Comment by codebytere** [comment]
@chrilith https://github.com/electron/electron/pull/11959 got merged,but didn't get backported into `2.0.0`...i'll key that up for backporting rn!

**Comment by chrilith** [comment]
@codebytere yeah, what I thought reading the history of the PR. Will wait for the merge 👍 

Thanks!
